c8y software versions list error on RaspberryPi

I receive an error when trying to run the 'software versions list --software' command on a RaspberryPi.  Here is the script I'm using to set my session and run the command:

```
#!/bin/bash

start_session(){
    C8Y_HOST=""
    C8Y_TENET=""
    C8Y_USERNAME=""
    C8Y_PASSWORD=""
    
    echo "Setting go-c8y-cli source"
    source "/etc/go-c8y-cli/shell/c8y.plugin.sh"

    echo "Setting go-c8y-cli session"
    c8y sessions create \
        --host "$C8Y_TENET" \
        --username "$C8Y_USERNAME" \
        --password "$C8Y_PASSWORD" \
        --description "Deploy Testing" \
        --type dev

    echo "Starting go-c8y-cli session"
    set-session

    echo "Setting mode.enableCreate TRUE"
    c8y settings update mode.enableCreate 'true'
}
 install_software(){
     local name="$1"             
     
     echo "Installing software: $name"
 }

clear_session(){
    c8y sessions clear
}

#
# Main
#
start_session

if [ $1 == P ]; then
    LATEST_VERSION=$( c8y software versions list -n --software asdm_algorithm_stack_P --includeAll --select "*.lastUpdated " -o csv | sort -Vr | head -1 )
elif [ $1 == Q ]; then
    echo "Setting Q software install"
else
    echo "No software release type set"
    exit 1
fi

c8y software versions list --software 740
     
#install_software "$LATEST_VERSION"
clear_session
```

This is the command prompt output when running the script:

```
./c8y_asdm_install.sh Q
Setting go-c8y-cli source
Setting go-c8y-cli session
/home/pi/.cumulocity/secasdm.us.cumulocity.com-rebecca.afzal@schindler.com.json
Starting go-c8y-cli session
---------------------  Cumulocity Session  ---------------------

    path: /home/pi/.cumulocity/secasdm.us.cumulocity.com-rebecca.afzal@schindler.com.json


host         : secasdm.us.cumulocity.com
tenant       : secasdm
username     : rebecca.afzal@schindler.com

Setting mode.enableCreate TRUE
Updated session file /home/pi/.cumulocity/secasdm.us.cumulocity.com-rebecca.afzal@schindler.com.json
Setting Q software install
panic: unaligned 64-bit atomic operation

goroutine 11 [running]:
runtime/internal/atomic.panicUnaligned()
	/opt/hostedtoolcache/go/1.16.13/x64/src/runtime/internal/atomic/unaligned.go:8 +0x24
runtime/internal/atomic.Xadd64(0x27f65f4, 0x1, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.16.13/x64/src/runtime/internal/atomic/asm_arm.s:233 +0x14
github.com/reubenmiller/go-c8y-cli/pkg/iterator.(*SliceIterator).GetNext(0x27f65e8, 0x0, 0x0, 0x24648, 0xa6d6dcc8, 0x1c17c, 0x19d44, 0x2884008)
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/iterator/slice.go:16 +0x34
github.com/reubenmiller/go-c8y-cli/pkg/c8yfetcher.(*EntityIterator).GetNext(0x280a210, 0x765901, 0x280a210, 0x9105fc, 0x280a210, 0xa6cbf201, 0x0, 0x1c418)
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/c8yfetcher/c8yfetcher.go:234 +0x34
github.com/reubenmiller/go-c8y-cli/pkg/flags.(*StringTemplate).Execute(0x27bc570, 0x5e700, 0x0, 0x0, 0x0, 0x245fd18, 0x0, 0x2884000, 0x0, 0x1, ...)
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/flags/stringtemplate.go:77 +0x2e8
github.com/reubenmiller/go-c8y-cli/pkg/flags.(*StringTemplate).GetNext(0x27bc570, 0x245fd18, 0x0, 0x0, 0x1, 0x2884000, 0x0, 0x1)
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/flags/stringtemplate.go:100 +0x38
github.com/reubenmiller/go-c8y-cli/pkg/requestiterator.(*RequestIterator).GetNext(0x28246c0, 0x7d9a97, 0x19, 0x2432e4c, 0x1, 0x1)
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/requestiterator/requestiterator.go:86 +0xc6c
github.com/reubenmiller/go-c8y-cli/pkg/worker.(*Worker).runBatched.func1(0x27454c0, 0x2823110, 0x28102e0, 0x28246c0, 0x0, 0x0, 0x2745500, 0x0, 0x0, 0x1, ...)
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/worker/worker.go:223 +0x100
created by github.com/reubenmiller/go-c8y-cli/pkg/worker.(*Worker).runBatched
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/worker/worker.go:215 +0x368

```

I am able to run 'c8y software list' without  getting the error.

--- a/pkg/iterator/func.go
+++ b/pkg/iterator/func.go
@@ -7,9 +7,9 @@ import (
 
 // FuncIterator is generic iterator which executes a function on every iteration
 type FuncIterator struct {
-	next         func(int64) (string, error)
+	currentIndex int64 // access atomically (must be defined at the top)
 	endIndex     int64
-	currentIndex int64
+	next         func(int64) (string, error)
 }
 
 // GetNext will count through the values and return them one by one

--- a/pkg/iterator/repeat.go
+++ b/pkg/iterator/repeat.go
@@ -7,9 +7,9 @@ import (
 
 // RepeatIterator is an empty iterator that always returns no value
 type RepeatIterator struct {
-	value        string
+	currentIndex int64 // access atomically (must be defined at the top)
 	endIndex     int64
-	currentIndex int64
+	value        string
 }
 
 // GetNext will count through the values and return them one by one

--- a/pkg/iterator/slice.go
+++ b/pkg/iterator/slice.go
@@ -7,8 +7,8 @@ import (
 
 // SliceIterator is iterates over a given array
 type SliceIterator struct {
+	currentIndex int64 // access atomically (must be defined at the top)
 	values       []string
-	currentIndex int64
 }
 
 // GetNext will count through the values and return them one by one

